### PR TITLE
chore(deps) bump OpenResty compatibility

### PR DESCRIPTION
**Pull request by thibaultcha**
### Summary

Bump OpenResty compatibility to `1.11.2.3` and fix a few issues along the way, especially regarding the recent LuaJIT string hashing updates.

### Full changelog

* bump OpenResty compat up to `1.11.2.3`
* update JWT tests fixtures after OR `1.11.2.3` bump
* hard-code plugins priorities as of `1.11.2.2` to the best of my knowledge

### TODO

* [x]  waiting on a release from Lapis to ensure routes prioritization according to the number of variables, for the `upstreams/{upstream}/targets/active` route.

### CODE REVIEW

**Comment by thibaultcha** [comment]
This is still waiting for a Lapis release. Lapis is not getting a release because apparently it needs to be stable on cqueues or something first. Yet it is currently *broken on OpenResty 1.11.2.2 for now, due to some LuaJIT/lua-cjson incompatibilities.

At this point, I think we need to consider forking it and uploading the current master branch of Lapis onto our own LuaRocks `mashape-lapis` rock.

File: spec/helpers.lua [hunk 1]
```diff
@@ -26,6 +26,54 @@ log.set_lvl(log.levels.quiet) -- disable stdout logs in tests
...
+  str = string.match(str, "^%s*(%S.-%S*)%s*$")
+  if not str then
+    return ""
+  end
+
+  local level  = math.huge
+  local prefix = ""
+  local len
+
+  for pref in str:gmatch("\n(%s+)") do
```
Comment (thibaultcha) [inline spec/helpers.lua]: Slightly inconsistent usage of the `string` library... I could update that.

**Comment by Tieske** [comment]
since Penlight got an update, should we reenable 52 compatibility? see also https://github.com/Mashape/kong/pull/2375#issuecomment-315886420

**Comment by thibaultcha** [comment]
> since Penlight got an update, should we reenable 52 compatibility? see also #2375 (comment)

Yes - although I think this should be part of another PR (I will submit it soon after merging this one)

**Pull request merged**